server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3ObjectStorage.java

```diff
@@ -182,108 +161,67 @@ private File createTempFile(Blob blob) throws IOException {
-                sink.error(new UploadFailedException());
-                transferManager.shutdownNow();
-            } else if (progressEvent.getEventType() == ProgressEventType.TRANSFER_COMPLETED_EVENT) {
-                sink.success();
-                transferManager.shutdownNow();
-            }
-
-            // ignore other events
-        }
-
-        private void closeQuietly(InputStream inputStream) {
-            try {
-                inputStream.close();
-            } catch (IOException e) {
-                LOGGER.error("Cannot close resource", e);
-            }
         }
 
         private void createBucket(ObjectStorageBucketName bucketName, AwsS3AuthConfiguration configuration) {
-            getS3Client(configuration, getClientConfiguration())
-                .createBucket(bucketName.asString());
-        }
-
-        private boolean failOnNoSuchBucket(Throwable th, ObjectStorageBucketName bucketName) {
-            return th instanceof UploadFailedException
-                && !doesBucketExist(bucketName);
-        }
-
-        private boolean failOnExistingBucket(Throwable th, ObjectStorageBucketName bucketName) {
-            return th instanceof UploadFailedException
-                && doesBucketExist(bucketName);
-        }
-
-        private boolean doesBucketExist(ObjectStorageBucketName bucketName) {
-            return getS3Client(configuration, getClientConfiguration())
-                .doesBucketExistV2(bucketName.asString());
-        }
-
-        private TransferManager getTransferManager() {
-            ClientConfiguration clientConfiguration = getClientConfiguration();
-            AmazonS3 amazonS3 = getS3Client(configuration, clientConfiguration);
-
-            return TransferManagerBuilder
-                    .standard()
-                    .withS3Client(amazonS3)
-                    .withMultipartUploadThreshold(MULTIPART_UPLOAD_THRESHOLD.getValue())
-                    .withExecutorFactory(() -> executorService)
-                    .withShutDownThreadPools(DO_NOT_SHUTDOWN_THREAD_POOL)
-                    .build();
-        }
-
-        private static AmazonS3 getS3Client(AwsS3AuthConfiguration configuration, ClientConfiguration clientConfiguration) {
-            return AmazonS3ClientBuilder
-                    .standard()
-                    .withClientConfiguration(clientConfiguration)
-                    .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(configuration.getAccessKeyId(), configuration.getSecretKey())))
-                    .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(configuration.getEndpoint(), null))
-                    .build();
-        }
+            getS3AsyncClient(configuration)
+                .map(client -> Mono.fromFuture(client
+                    .createBucket(CreateBucketRequest.builder()
+                        .bucket(bucketName.asString())
+                        .build())))
+                .block();
```

Comment:
why would we want to block here?

Reply:
this method is used in Retry. `Retry.doOnRetry(any -> creatingBucket())` just supports to pass a consumer into. So on I have to do it synchronously